visualization/src/pcl_visualizer.cpp

```diff
@@ -122,55 +122,58 @@
   #undef far
 #endif
 
-int
+auto
 pcl::visualization::details::fillCells(std::vector<int>& lookup, const std::vector<pcl::Vertices>& vertices, vtkSmartPointer<vtkCellArray> cell_array)
 {
-  int idx = 0;
 #ifdef VTK_CELL_ARRAY_V2
   if (!lookup.empty())
   {
-    for (std::size_t i = 0; i < vertices.size(); ++i, ++idx)
+    for (const auto& verti : vertices)
     {
-      std::size_t n_points = vertices[i].vertices.size();
+      std::size_t n_points = verti.vertices.size();
       cell_array->InsertNextCell(n_points);
-      for (std::size_t j = 0; j < n_points; j++, ++idx)
-        cell_array->InsertCellPoint(lookup[vertices[i].vertices[j]]);
+      for (const auto& vertj : verti.vertices)
+        cell_array->InsertCellPoint(lookup[vertj]);
     }
   }
   else
   {
-    for (std::size_t i = 0; i < vertices.size(); ++i, ++idx)
+    for (const auto& verti : vertices)
     {
-      std::size_t n_points = vertices[i].vertices.size();
+      std::size_t n_points = verti.vertices.size();
       cell_array->InsertNextCell(n_points);
-      for (std::size_t j = 0; j < n_points; j++, ++idx)
-        cell_array->InsertCellPoint(vertices[i].vertices[j]);
+      for (const auto& vertj : verti.vertices)
+        cell_array->InsertCellPoint(vertj);
     }
   }
 #else
   vtkIdType* cell = cell_array->WritePointer(vertices.size(), vertices.size() * (max_size_of_polygon + 1));
 
   if (!lookup.empty())
   {
-    for (std::size_t i = 0; i < vertices.size(); ++i, ++idx)
+    for (const auto& verti : vertices)
     {
-      std::size_t n_points = vertices[i].vertices.size();
+      std::size_t n_points = verti.vertices.size();
       *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, ++idx)
-        *cell++ = lookup[vertices[i].vertices[j]];
+      for (const auto& vertj : verti.vertices)
+        *cell++ = lookup[vertj];
     }
   }
   else
   {
-    for (std::size_t i = 0; i < vertices.size(); ++i, ++idx)
+    for (const auto& verti : vertices)
     {
-      std::size_t n_points = vertices[i].vertices.size();
+      std::size_t n_points = verti.vertices.size();
       *cell++ = n_points;
-      for (std::size_t j = 0; j < n_points; j++, ++idx)
-        *cell++ = vertices[i].vertices[j];
+      for (const auto& vertj : verti.vertices)
+        *cell++ = vertj;
     }
   }
 #endif
+
+  const auto idx = vertices.size() + std::accumulate(vertices.begin(), vertices.end(), 0,
```

Comment:
0 will be by default `int`. You need to cast it to the intended type (say vtkIdType)

Reply:
This will have idx as `int` since 0 is `int`